apps/web/src/views/Swap/SmartSwap/components/TransactionConfirmSwapContent.tsx

```diff
@@ -0,0 +1,126 @@
+import SwapModalFooter from '../../components/SwapModalFooter'
+import SwapModalHeader from '../../components/SwapModalHeader'
+import StableSwapModalHeader from '../../StableSwap/components/StableSwapModalHeader'
+import {
+  computeSlippageAdjustedAmounts as computeSlippageAdjustedAmountsWithSmartRouter,
+  computeTradePriceBreakdown as computeTradePriceBreakdownWithSmartRouter,
+} from '../utils/exchange'
+
+/**
+ * Returns true if the trade requires a confirmation of details before we can submit it
+ * @param tradeA trade A
+ * @param tradeB trade B
+ */
+function tradeMeaningfullyDiffers(tradeA: ITrade, tradeB: ITrade): boolean {
+  return (
+    tradeA.tradeType !== tradeB.tradeType ||
+    !tradeA.inputAmount.currency.equals(tradeB.inputAmount.currency) ||
+    !tradeA.inputAmount.equalTo(tradeB.inputAmount) ||
+    !tradeA.outputAmount.currency.equals(tradeB.outputAmount.currency) ||
+    !tradeA.outputAmount.equalTo(tradeB.outputAmount)
+  )
+}
+
+interface TransactionConfirmSwapContentProps {
+  trade: TradeWithStableSwap<Currency, Currency, TradeType> | undefined
+  originalTrade: ITrade | undefined
+  onAcceptChanges: () => void
+  allowedSlippage: number
+  onConfirm: () => void
+  recipient: string
+  currencyBalances: {
+    INPUT?: CurrencyAmount<Currency>
+    OUTPUT?: CurrencyAmount<Currency>
+  }
+  isStable: boolean
+}
+
+const TransactionConfirmSwapContent = ({
+  trade,
+  originalTrade,
+  onAcceptChanges,
+  allowedSlippage,
+  onConfirm,
+  recipient,
+  currencyBalances,
+  isStable,
+}: TransactionConfirmSwapContentProps) => {
+  const showAcceptChanges = useMemo(
+    () => Boolean(trade && originalTrade && tradeMeaningfullyDiffers(trade, originalTrade)),
+    [originalTrade, trade],
+  )
+
+  const slippageAdjustedAmounts = useMemo(
+    () => computeSlippageAdjustedAmountsWithSmartRouter(trade, allowedSlippage),
+    [trade, allowedSlippage],
+  )
+  const { priceImpactWithoutFee } = useMemo(() => computeTradePriceBreakdownWithSmartRouter(trade), [trade])
+
+  const isEnoughInputBalance = useMemo(() => {
+    if (trade?.tradeType !== TradeType.EXACT_OUTPUT) return null
+
+    const isInputBalanceExist = !!(currencyBalances && currencyBalances[Field.INPUT])
+    const isInputBalanceBNB = isInputBalanceExist && currencyBalances[Field.INPUT].currency.isNative
+    const inputCurrencyAmount = isInputBalanceExist
+      ? isInputBalanceBNB
+        ? maxAmountSpend(currencyBalances[Field.INPUT])
+        : currencyBalances[Field.INPUT]
+      : null
+    return inputCurrencyAmount && slippageAdjustedAmounts && slippageAdjustedAmounts[Field.INPUT]
+      ? inputCurrencyAmount.greaterThan(slippageAdjustedAmounts[Field.INPUT]) ||
+          inputCurrencyAmount.equalTo(slippageAdjustedAmounts[Field.INPUT])
+      : false
+  }, [currencyBalances, trade, slippageAdjustedAmounts])
+
+  const modalHeader = useCallback(() => {
+    const SwapModalHead = isStable ? StableSwapModalHeader : SwapModalHeader
```

Comment:
We also don't need to worry about if it's stable swap or not? Cuz if this component is used, it means that the trade must include stable swap, otherwise it would fallback to the old commit button

Reply:
I think maybe we should duplicate the `SwapModalHead` as well and use it directly. Of course the best thing would be decouple the modal header component with the trade type, but let's just duplicate for now